tests/integration_test.rs

```diff
@@ -19,3 +19,17 @@ fn test_no_panic_if_empty_string() {
     perform_cracking("", config);
     assert_eq!(true, true);
 }
+
+// To test if max depth limit is working, we set max_depth in the global config
+// Due to config being global, we don't want it to interfere with other tests, or vice-versa
```

Comment:
At Monzo we have a function that "cleans" the program so it goes back to default settings, you call it at the start of every test which you might think is "dirty" (with differing configs)

Reply:
It isn't not possible to change the values inside OnceCell. If we want to mutate it, we would need to wrap it in a mutex like `OnceCell<Mutex<Config>>`